hadoop-ozone/s3gateway/src/test/java/org/apache/hadoop/ozone/s3/endpoint/TestMultipartUploadWithCopy.java

```diff
@@ -208,6 +218,73 @@ public void testMultipartIfModifiedSince() throws Exception {
     }
   }
 
+  // Test operations with 2 datestrings; one valid, the other invalid;
+  // confirm the invalid one is ignored and the valid precondition tested
```

Comment:
Thank you for adding this, we should have 12 test cases or more, 4 based on true and false condition for timestamps that are valid, 4 with either one of the time stamps is null and the other true or false, 4 for ether timestamp in the future and other time stamp true or false but valid and finally actual bad input.

Reply:
Will do, Thanks @kerneltime!

Reply:
hey @kerneltime I've added the tests you suggested to s3/endpoint/TestMultipartUploadWithCopy.java.  There ended up being 25 different combinations total.  I had to restructure the tests a bit to reduce the repeated boilerplate.  Hopefully those changes won't be too hard to follow.


Here are just the changes since your review, if that makes it easier:
https://github.com/GeorgeJahad/ozone/compare/3e7edd84a1a7cbf3e0c8a95c05ae971be171ffa1..f567f523bd12f88ff6db96c5cfdfa6d0a853399a#diff-bd6abd262ed8e24eaeea5cb0e7c2a1a128411b284f9f27dae1ad47be6717a940

Thanks again!